courses/utils/utils.py

```diff
@@ -0,0 +1,16 @@
+from django.utils.text import slugify
+
+
+def get_unique_slug(model_instance, slugable_field_name, slug_field_name):
+    slug = slugify(getattr(model_instance, slugable_field_name))
+    unique_slug = slug
+    extension = 1
+    ModelClass = model_instance.__class__
```

Comment:
I feel this is unnecessarily complex. 
- https://keyerror.com/blog/automatically-generating-unique-slugs-in-django
- https://docs.djangoproject.com/en/2.0/ref/utils/#django.utils.text.slugify
- KISS